packages/11ty/helpers/page-filters.js

```diff
@@ -0,0 +1,35 @@
+/**
+ * Returns `true` if page should be built
+ * for the current output type defined in `env.QUIRE_OUTPUT`
+ * @param  {Object} data Eleventy `page.data` object
+ * @return {Boolean}
+ */
+const currentOutputFilter = function ({ outputs }) {
+  return outputs !== 'none' &&
+    outputs !== false &&
+    (!outputs ||
+      Array.isArray(outputs) && outputs.includes(process.env.QUIRE_OUTPUT) ||
+      outputs === process.env.QUIRE_OUTPUT)
+}
+
+/**
+ * Returns `true` if page should be included in the menu
+ * @param  {Object} page Eleventy `page` object
+ * @return {Boolean}
+ */
+const menuFilter = function(data) {
+  const { menu, type } = data
+  return (currentOutputFilter(data) || menu === true) && menu !== false && type !== 'data'
+}
```

Comment:
In `.eleventy.js`
```
eleventyConfig.addCollection('menu', pageFilters.menuFilter(collectionApi))
```
though it may be more encapsulated to init the `pageFilters` module with `collectionApi` as an arg, like
```
eleventyConfig.addCollection('menu', pageFilters(collectionApi).menuFilter())
```
In `pageFilters`, using the second approach above something like,
```
const pages = const pages = collectionApi.getAll()
const menuFilter = function() {
  const currentOutput = currentOutputFilter(pages)
  return // filtered menu items
}
```

Reply:
Leaving for further refactoring - our use of collections and filters seems in flux still, not sure what the best implementation will be.